zerver/views/registration.py

```diff
@@ -861,7 +860,9 @@ def realm_redirect(request: HttpRequest, next: str = REQ(default="")) -> HttpRes
         if form.is_valid():
             subdomain = form.cleaned_data["subdomain"]
             realm = get_realm(subdomain)
-            redirect_to = get_safe_redirect_to(next, realm.uri)
+            redirect_to = append_url_query_string(
+                realm.uri + settings.HOME_NOT_LOGGED_IN, urlencode({REDIRECT_FIELD_NAME: next})
```

Comment:
Let's use `urljoin` instead of `ealm.uri + settings.HOME_NOT_LOGGED_IN` string addition perhaps?

Reply:
And I think the redirect URL is nicer without a dangling, empty `next=` if there is no next value provided, so perhaps the query string should only happen `if next`?

I think that answers the question in https://github.com/zulip/zulip/pull/23362#issuecomment-1304432687 now that I look at it again